aries_cloudagent/vc/data_integrity/cryptosuites/eddsa_jcs_2022.py

```diff
@@ -0,0 +1,201 @@
+        assert (
+            proof_config["cryptosuite"] == "eddsa-jcs-2022"
+        ), 'Expected proof.cryptosuite to be "eddsa-jcs-2022'
+
+        if "created" in proof_config:
+            # TODO assert proper [XMLSCHEMA11-2] dateTimeStamp string
+            assert proof_config[
+                "created"
+            ], "Expected proof.created to be a [XMLSCHEMA11-2] dateTimeStamp string."
+
+        if "expires" in proof_config:
+            # TODO assert proper [XMLSCHEMA11-2] dateTimeStamp string
+            assert proof_config[
+                "expires"
+            ], "Expected proof.expires to be a [XMLSCHEMA11-2] dateTimeStamp string."
+
+        return self._canonicalize(proof_config)
+
+    def transformation(self, unsecured_document: dict, options: dict):
+        """Transformation algorithm.
+
+        https://www.w3.org/TR/vc-di-eddsa/#transformation-eddsa-jcs-2022.
+        """
+        assert (
+            options["type"] == "DataIntegrityProof"
+        ), "Expected proof.type to be `DataIntegrityProof`"
+        assert (
+            options["cryptosuite"] == "eddsa-jcs-2022"
+        ), "Expected proof.cryptosuite to be `eddsa-jcs-2022`"
+
+        return self._canonicalize(unsecured_document)
+
+    def hashing(self, transformed_document, canonical_proof_config):
+        """Hashing algorithm.
+
+        https://www.w3.org/TR/vc-di-eddsa/#hashing-eddsa-jcs-2022.
+        """
+        return (
+            sha256(canonical_proof_config).digest()
+            + sha256(transformed_document).digest()
+        )
+
+    async def proof_serialization(self, hash_data: bytes, options: dict):
+        """Proof Serialization Algorithm.
+
+        https://www.w3.org/TR/vc-di-eddsa/#proof-serialization-eddsa-jcs-2022.
+        """
+        # If the verification method is a did:key: URI,
+        # we derive the signing key from a multikey value
+        if options["verificationMethod"].startswith("did:key:"):
+            multikey = options["verificationMethod"].split("#")[-1]
+            key_info = await self.key_manager.from_multikey(multikey)
+
+        # Otherwise we derive the signing key from a kid
+        else:
+            key_info = await self.key_manager.from_kid(options["verificationMethod"])
+
+        return await self.wallet.sign_message(
+            message=hash_data,
+            from_verkey=self.key_manager._multikey_to_verkey(key_info["multikey"]),
+        )
+
+    def _canonicalize(self, data: dict):
+        """Json canonicalization."""
+        return canonicaljson.encode_canonical_json(data)
+
+    async def _get_multikey(self, kid: str):
+        """Derive a multikey from the verification method."""
+
+        # If verification method is a did:key URI,
+        # we derive the multikey directly from the value.
+        if kid.startswith("did:key:"):
+            return kid.split("#")[-1]
+
+        # Otherwise we resolve the verification method and extract the multikey.
+        else:
```

Comment:
Why bother special casing did:key? "Resolution," as it were, for did:key is cheap so might as well handle it consistently with other DID Methods, IMO.

Reply:
Resolvers are inconsistent with how they return a `did:key:`, we specifically want a multikey value here. For example, the uniresolver will return a JWK from a `did:key:` resolving step. This cryptosuite defines a multikey as the verification method and this is what the manager will be expecting. did key is an easy shortcut for this since it's literally the multikey value.

Reply:
We have control over the resolvers in ACA-Py; we can make it always return a multikey, even if we have registered the uniresolver as a resolver in ACA-Py. "Native" resolvers are given preference over "remote" resolvers.

Reply:
got rid of this check, we are now resolving all verification method